miden-tx/src/host/mod.rs

```diff
@@ -386,4 +386,369 @@ impl<A: AdviceProvider> Host for TransactionHost<A> {
 
         Ok(HostResponse::None)
     }
+
+    fn on_assert_failed<S: ProcessState>(&mut self, process: &S, err_code: u32) -> ExecutionError {
+        match err_code {
+            131072 => ExecutionError::FailedAssertion { // 0x00020000: ERR_FAUCET_RESERVED_DATA_SLOT
```

Comment:
Can we use constants here? This way, we won't have to have the comment `0x00020000: ERR_FAUCET_RESERVED_DATA_SLOT`

Reply:
I thought about that, but in that case we will have additional ~100 lines of constants just to associate the `u32` error code with __short__ constant name, which is not used anywhere else. So that's why I decided to make mapping directly from error code to the long error message -- comment in that case doing the same job as constants but, allowing not to create another enum or a long constants list and making solution more compact. 
But that's just my thoughts, so I can implement it if you say so.

Reply:
I think it might be good to have the constants done explicitly (and use the same values as in MASM code) - this way, they would be easier to find.

btw, another way to look up the messages is in a map. Basically, we can declare a constant array:
```Rust
const KERNEL_ERRORS: [(u32, &str); <num errors>] = [
    (ERR_FAUCET_RESERVED_DATA_SLOT, "For faucets the slot FAUCET_STORAGE_DATA_SLOT is reserved and can not be used with set_account_item"),
    ...
];
```
And then in the `TransactionHost` constructor we could create a `BTreeMap` out of this array. This map can then be used inside `on_assert_failed()` to find the right message for a given code).

This would probably be less code. The downside is that we can't construct dynamic messages this way (which I don't think we currently do).